src/Data/PQueue/Prio/Internals.hs

```diff
@@ -707,64 +700,43 @@ mapMWithKey f = go empty
           let !acc' = insertMax' k b acc
           go acc' q'
 
+-- | Natural numbers revealing whether something is 'Zero' or 'Succ'.
+type Ranky = Nattish Zero Succ
```

Comment:
Is `Nattish` only used for this? If so, wouldn't it be more clear to only define `Ranky`?

Reply:
I would like to consider using it for the plain queues too.